### PR TITLE
test: check PULL_REQUEST_BAZEL_TARGETS

### PR DESCRIPTION
**Pull request by basvandijk**
This adds the CI job "CI Main / Check PULL_REQUEST_BAZEL_TARGETS" which calls:
```
ci/bazel-scripts/targets.py check
```
which checks that the `PULL_REQUEST_BAZEL_TARGETS` file:
* can be read and parsed.
* each pattern matches at least one file tracked by git.
* each pattern has at least one explicit target.
* each bazel target is valid and exists.

When errors are introduced (like [these](https://github.com/dfinity/ic/pull/6333/commits/04af8ef68dfcb1fbb4a49279c1d2be7cf6cafc1e) for example) the output looks like [this](https://github.com/dfinity/ic/actions/runs/17052439782/job/48343318748?pr=6333#step:5:1058).

### CODE REVIEW

File: .github/workflows-source/ci-main.yml [hunk 1]
```diff
@@ -640,3 +638,32 @@ jobs:
...
+    name: Check PULL_REQUEST_BAZEL_TARGETS
+    <<: *dind-small-setup
+    steps:
+      - <<: *checkout
+      - name: Filter files relevant for checking PULL_REQUEST_BAZEL_TARGETS
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3
+        id: filter
+        if : |
+          github.event_name == 'pull_request' ||
+          github.event_name == 'merge_group'
```
Comment (cgundy) [inline .github/workflows-source/ci-main.yml]: I thin this should be part of the workflow `ci-pr-only`, then it's only triggered pull requests and merge groups.

File: .github/workflows-source/ci-main.yml [hunk 2]
```diff
@@ -640,3 +638,32 @@ jobs:
...
+          filters: |
+            src:
+              - "**/*.bzl"
+              - "**/*.bazel"
+              - "PULL_REQUEST_BAZEL_TARGETS"
+              - "ci/bazel-scripts/targets.py"
+      - name: Check PULL_REQUEST_BAZEL_TARGETS
+        if: |
+          steps.filter.outputs.src == 'true' ||
+          github.event_name == 'schedule' ||
```
Comment (cgundy) [inline .github/workflows-source/ci-main.yml]: This logic doesn't quite make sense to me... you previously check if it's run on a PR or merge group, then check if it's run on a schedule?

File: .github/workflows-source/ci-main.yml (hunk 1 above)
Comment (basvandijk) [reply .github/workflows-source/ci-main.yml]: Is there a reason we shouldn't run it on pushes to master and during Release Testing?

File: .github/workflows-source/ci-main.yml (hunk 2 above)
Comment (basvandijk) [reply .github/workflows-source/ci-main.yml]: Note that this is the same logic as for the `cargo-build-release-linux` job above. In fact I copied that job to write this one.

So when in a PR the step is only run when certain files have been modified. 

It runs regardless otherwise.

Although I see both the step in `cargo-build-release-linux`and this step don't run when `github.event_name == "push"` which I just fixed. (I will submit a fix for `cargo-build-release-linux` later.

Thanks for makinge realise!

File: .github/workflows-source/ci-main.yml [hunk 3]
```diff
@@ -640,3 +638,32 @@ jobs:
...
+          github.event_name == 'workflow_dispatch' ||
+          github.event_name == 'workflow_call'
```
Comment (basvandijk) [inline .github/workflows-source/ci-main.yml]:
```suggestion
          github.event_name == 'workflow_call' ||
          GitHub.event_name == 'push'
```

Comment (basvandijk) [inline .github/workflows-source/ci-main.yml]:
```suggestion
          github.event_name == 'workflow_call' ||
          GitHub.event_name == 'push'
```

File: .github/workflows-source/ci-main.yml (hunk 2 above)
Comment (basvandijk) [reply .github/workflows-source/ci-main.yml]: Actually now that I think about it some more we should always run this step for every PR regardless of which change. Because any pattern in `PULL_REQUEST_BAZEL_TARGETS` could go invalid because of any change in a PR.

So I'll remove the `Filter files relevant for checking PULL_REQUEST_BAZEL_TARGETS` step above and simply remove the whole `if:` here.

File: .github/workflows-source/ci-main.yml (hunk 1 above)
Comment (cgundy) [reply .github/workflows-source/ci-main.yml]: Well yeah because we don't run linting in those scenarios either. We should only need to run it when a change is made to the file, I don't see a scenario where it passes all the checks on a PR then the file somehow changes and it fails on master. Or do you see that as a valid case?

Comment (basvandijk) [reply .github/workflows-source/ci-main.yml]: We do run linters on pushes to master. See [these linting tests](https://dash.dm1-idx1.dfinity.network/invocation/0e9eb18a-3b9b-4c49-8b1e-34ecd2201131?targetFilter=lint#targets) for a recent push to master for example.

You could make the same argument that for any test that runs on a PR you don't need to run it on master since it would be testing the same thing. However we choose to run these tests again just to be sure that a specific git commit is correct regardless of how it got there.

Comment (cgundy) [reply .github/workflows-source/ci-main.yml]: Well yeah I'm not sure then. Maybe we should have a discussion on which tests should run on CI-Main and which ones on CI-PR only. Because right now it seems a little random.

Comment (cgundy) [reply .github/workflows-source/ci-main.yml]: I'm just less of a fan of using complex logic within a CI job to determine when it should be run. It creates a lot more debugging steps to understand when a job was skipped vs why it was run. It's cleanest to follow the triggers at the top of the file. I recognize there will always be edge cases and we have a lot of other tests that do the same, but I think keeping the triggers simple when possible does help a lot.

Comment (basvandijk) [reply .github/workflows-source/ci-main.yml]: Yes good points! I've now simplified it to always run in any case which is the correct and simple thing to do.

Comment (cgundy) [reply .github/workflows-source/ci-main.yml]: Okay great, thanks for that! :)

**Review by cgundy — APPROVED** [review]
